### PR TITLE
correct omp dependency for mkl-dnn

### PR DESCRIPTION
**Pull request by gujinghui**
The motivational of this PR is to enforce mkldnn to use the same omp version of caffe2 framework.
Meanwhile, do not change other assumptions within mkldnn.

Previously, the MKL_cmake_included is set in caffe2 in order to disable omp seeking in mkldnn.
But, with such change, mkldnn has no chance to adapt for mkl found by caffe2.
Then, some building flags of mkl will be not set in mkldnn.
For example, USE_MKL, USE_CBLAS, etc.

In this PR, we enforce set the MKLIOMP5LIB for mkldnn according to caffe2, and tell the mkl root path in MKLROOT for mkldnn. Then, mkldnn is built as expected.

### CODE REVIEW

**Comment by gujinghui** [comment]
@yinghai
please help review.

File: cmake/Modules/FindMKLDNN.cmake [hunk 1]
```diff
@@ -43,6 +43,25 @@ LIST(APPEND MKLDNN_INCLUDE_DIR ${IDEEP_INCLUDE_DIR})
 IF(MKL_FOUND)
   LIST(APPEND MKLDNN_LIBRARIES ${MKL_LIBRARIES})
   LIST(APPEND MKLDNN_INCLUDE_DIR ${MKL_INCLUDE_DIR})
+  # The OMP-related variables of MKL-DNN have to be overwritten here,
+  # if MKL is using, and the OMP version is defined by MKL.
```
Comment (yinghai) [inline cmake/Modules/FindMKLDNN.cmake]: `used`

File: cmake/Modules/FindMKLDNN.cmake [hunk 2]
```diff
@@ -43,6 +43,25 @@ LIST(APPEND MKLDNN_INCLUDE_DIR ${IDEEP_INCLUDE_DIR})
...
+  # MKL_LIBRARIES_xxxx_LIBRARY is defined by MKL.
+  # INTEL_MKL_DIR gives the MKL root path.
+  IF (INTEL_MKL_DIR)
+    SET(MKLROOT ${INTEL_MKL_DIR})
+    IF(WIN32)
+      SET(MKLIOMP5DLL ${MKL_LIBRARIES_libiomp5md_LIBRARY} CACHE STRING "Overwrite MKL-DNN omp dependency" FORCE)
+    ELSE(WIN32)
+      IF (MKL_LIBRARIES_gomp_LIBRARY)
```
Comment (yinghai) [inline cmake/Modules/FindMKLDNN.cmake]: Where is this flag set?

**Review by yinghai — CHANGES_REQUESTED** [review]
Please explain a bit more about the motivational of this PR. Specifically, why OMP related variable has to be overwritten.

File: cmake/Modules/FindMKLDNN.cmake (hunk 1 above)
Comment (gujinghui) [reply cmake/Modules/FindMKLDNN.cmake]: fixed

File: cmake/Modules/FindMKLDNN.cmake (hunk 2 above)
Comment (gujinghui) [reply cmake/Modules/FindMKLDNN.cmake]: MKL_LIBRARIES_gomp_LIBRARY is set by FindMKL.cmake

**Comment by gujinghui** [comment]
@yinghai 

The motivational of this PR is to enforce mkldnn to use the same omp version of caffe2 framework.
Meanwhile, do not change other assumptions within mkldnn.

Previously, the MKL_cmake_included  is set in caffe2 in order to disable omp seeking in mkldnn.
But, with such change, mkldnn has no chance to adapt for mkl found by caffe2.
Then, some building flags of mkl will be not set in mkldnn.
For example, USE_MKL, USE_CBLAS, etc.

In this PR, we enforce set the MKLIOMP5LIB for mkldnn according to caffe2, and tell the mkl root path in MKLROOT for mkldnn. Then, mkldnn is built as expected.

**Review by yinghai — APPROVED** [review]

**Review by soumith — APPROVED** [review]
lgtm, go ahead and ship it @yinghai

**Review by orionr — APPROVED** [review]
Looks good! I would eventually like to replace all the `ELSE (XXX)` with `ELSE()` and `ENDIF (XXX)` with `ENDIF()` since it's more modern CMake syntax, but  not something to change right now.

**Pull request closed without merging**